scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,175 @@
+--- | ---
+`dev` | `master`
+`v1` | `release/4.6.0`
+
+If in 4.6 the bundle version was updated to:
+
+```
+$ cat canner/VULNERABILITY_VERSION
+v2
+```
+
+On 4.6.0 release cut, the final mapping would be:
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `release/4.5.0`
+`v2` | `release/4.6.0`
+
+#### Offline mode
+
+Per-release offline bundles will continue to be published.  They will be versioned by StackRox releases, but they will not be based on Y-stream (`4.5`) but on Z-stream patch releases (`4.5.0`).  The bundle will contain the correct vulnerability bundle for that particular Z-stream release.  The `manifest.json` file will continue to have a "version" attribute pointing out the Y-stream for that offline bundle, but two other fields will be added: "vulnerability_version" and "release_versions".  For example, for patch release `4.6.1` the offline bundle could be:
+
+```
+{
+  "version": "4.6",
+  "created": "2024-08-20T21:00:30+00:00",
+  "release_versions": ["4.6.0", "4.6.1", ...],
+  "vulnerability_version": "v1"
+}
+```
+
+The central definitions handler will validate the offline bundle POST against `release_versions` to ensure the running Central version is supported.  And, when consuming the offline bundle, against "vulnerability_version", guaranteeing the bundle supports the Scanner requested schema version.
+
+We will continue to offer the latest offline bundles.  They will contain all offline bundles versioned by schema, and the existing `4.5` bundle to continue to be added for backward compatibility.  Example:
+
+```
+unzip -l scanner-vuln-updates.zip                                                                             [~]
+Archive:  scanner-vuln-updates.zip
+  Length      Date    Time    Name
+---------  ---------- -----   ----
+171516200  2024-08-20 21:00   scanner-defs.zip
+   248088  2024-08-20 21:00   k8s-istio.zip
+191673607  2024-08-20 21:00   scanner-v4-defs-v1.zip
+191673030  2024-08-20 21:00   scanner-v4-defs-v2.zip
+191673030  2024-08-20 21:00   scanner-v4-defs-4.5.zip
+---------                     -------
+363437895                     3 files
+```
+
+The `roxctl scanner download-db` will continue to support downloading the correct bundle per release using the `--version` command line option.
+
+### Ingesting versioned bundles
+
+Scanner will know the vulnerability version it requires based on hard-coded information added during build time.  A file at `scanner/VULNERABILITY_VERSION` will determine the version on release builds and nightlies.  On non-release builds this will hard-coded to "dev".
+
+The definitions handler in Central can be kept the same.  It already supports arbitrary versions in the GET request from Scanner.  [The check for nightlies](https://github.com/stackrox/stackrox/blob/e332cd19a639f59d9931414a4f4e561981396dad/central/scannerdefinitions/handler/handler.go#L661) and other per-release checks in the ingestion path are unnecessary and will be removed.
```

Comment:
The hardcoding for nightlies exists because nightlies are release builds, and we never published vuln data for the particular nightly tag. With the new schema, removing this check will be ok, though, because it will be associated with something like `v1`, right?

Reply:
It will be associated with `scanner/VULNERABILITY_VERSION`, initially `v1`, yes. And `v1` should be available based on latest master branch build.